modules/ROOT/pages/api-extensions.adoc

```diff
@@ -0,0 +1,429 @@
+= Anypoint DataGraph API Extension Reference
+
+The following guide defines API extensions DataGraph. 
+
+
+== Object Type Extensions 
+
+This extension is used to mark that an object type has a primary key, and thus it can be used as reference or extension.
+
+This extension applies to `ObjectType`.
+
+=== Structure
+
+[%header%autowidth.spread]
+|===
+|Field |Type |Description
+|`entity` |Map |Marks an object type as entity.
+|`entity.key` |String |Array, must not be empty. Each item must reference a field that is a key of the entity.
+|===
+
+=== OAS Example
+
+----
+components:
+  schemas:
+    Customer:
+  	x-entity:
+    	  key: [id]
+  	properties:
+    	  name:
+          type: string
+    	  email:
+          type: string
+    	  id:
+          type: string
+    	  ...
+----
+
+=== RAML Example
+
+----
+types:
+  Customer:
+    (dg.entity):
+      key: [id]
+    properties:
+    	name:
+        type: string
+    	email:
+        type: string
+    	id:
+        type: string
+    	...
+----
+
+== Entity Reference
+
+This extension is used to generate a DataGraph _link_ from the properties of a type to another type. The link can be to a type that is not defined in the same API spec.
+
+This extension applies to `ObjectType`.
+
+=== Structure
+
+[%header%autowidth.spread]
+|===
+|Field |Type |Description
+|`entity-reference` |Array of objects |A list of all the references.
```

Comment:
General global comment - descriptions should be consistent regarding whether they start with an article like "a" or "the" or not. They are inconsistent in most of the tables.